packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -45,11 +46,17 @@ export interface SnapshotWithBlobs {
 /**
  * State saved by a container at close time, to be used to load a new instance
  * of the container to the same state
+ *
+ * This is very similar to {@link @fluidframework/protocol-definitions/internal#ISnapshot}, but the difference is
+ * that the blobs of ISnapshot of are ArrayBufferLike, while the blobs of this interface are serializable because
+ * they are already converted to string.
+ *
  * @internal
  */
 export interface IPendingContainerState extends SnapshotWithBlobs {
 	attached: true;
 	pendingRuntimeState: unknown;
+	loadedGroupIdSnapshots?: Record<string, ISnapshotInfo>;
```

Comment:
@anthony-murphy I have a feeling you may have opinions here, this is what we came up with as this would fit our correctness requirements.